core/domain/state_domain.py

```diff
@@ -277,6 +277,10 @@ class InteractionInstance(python_utils.OBJECT):
     # The default interaction used for a new state.
     _DEFAULT_INTERACTION_ID = None
 
+    # List of interactions storing HTML in the rule spec inputs field.
+    INTERACTIONS_STORING_HTML_RULE_SPEC_INPUT = [
```

Comment:
There should be a backend test that runs through all interactions and programmatically extracts all HTML fields, and verifies that the result is the same as the constant defined here. That way, we'll get an alert if this constant ever needs to be updated in the future.

Reply:
Done.

Reply:
This is getting closer to the right idea, though I think this is still more complex than necessary. Couldn't you just have one list of 3-tuples like ('ItemSelectionInput', 'Equals', 'x') for html strings, and a similar list for html lists? The idea is to just store a bunch of pointers to the relevant variables in the rules so that you can iterate over the lists and grab the data quickly -- and what you're doing at the moment seems a bit too complicated.

In fact, perhaps better: have a single dict mapping the tuples above to their types. That way, when you iterate through the interactions/rules, you can construct the tuple, and then see whether it's in the dict, and if so get the type directly and process it.